### DISCUSSION TITLE
JSON Schema

### DISCUSSION THREAD

**Discussion by garyisaac** [synthetichealth/synthea#1051] [General]
Thank you for your help. I wanted to know if there is a schema file for the JSON exporter. I am using a MS Power Automate and its parser needs wants a schema.

Thank you,
Gary

**Comment by jawalonoski** [comment]
Do you mean FHIR?

You can try the schema file linked from here: http://hl7.org/fhir/R4/downloads.html

**Comment by eedrummer** [comment]
There is no current JSON Schema for the JSON Exporter.

The JSON Export can really be thought of as a "dump" of the internal state of Synthea at the end of the simulation for a person. While I wouldn't expect wild changes in the structure, any changes to the Java objects that make up Synthea would ripple through to the export format.

**Reply by jawalonoski** [reply]
@eedrummer Let's get a wiki page up on the JSON exporter and highlight this and maybe some other issues with using it.

**Reply by eedrummer** [reply]
It was actually there, but never linked - https://github.com/synthetichealth/synthea/wiki/JSON-Export

It's now in the TOC and linked from the Records page. It was also updated to remove the not about only being on a branch.

**Comment by garyisaac** [comment]
It looks like EncounterModule.java defines the well-child periodicity so I would not be able to create a module using the Module Builder. I would have to go into the source code and create the periodicity schedule for ages under 19. Then Synthea would create encounters with data for the correct schedule. And, we could introduce some random anomalies to test missed, overdue, etc. encounters.

/**

* Recommended time between Wellness Visits by age of patient and whether

* they have chronic medications.

* @param person The patient.

* @param time The time of the encounter which we translate to age of patient.

* @return Recommended time between Wellness Visits in milliseconds

*/

public long recommendedTimeBetweenWellnessVisits(Person person, long time) {

int ageInYears = person.ageInYears(time);

boolean hasChronicMeds = !person.chronicMedications.isEmpty();


long interval; // return variable


if (ageInYears <= 3) {

int ageInMonths = person.ageInMonths(time);

if (ageInMonths <= 1) {

interval = Utilities.convertTime("months", 1);

} else if (ageInMonths <= 5) {

interval = Utilities.convertTime("months", 2);

} else if (ageInMonths <= 17) {

interval = Utilities.convertTime("months", 3);

} else {

interval = Utilities.convertTime("months", 6);

}

} else if (ageInYears <= 19) {

interval = Utilities.convertTime("years", 1);

} else if (ageInYears <= 39) {

interval = Utilities.convertTime("years", 3);

} else if (ageInYears <= 49) {

interval = Utilities.convertTime("years", 2);

} else {

interval = Utilities.convertTime("years", 1);
}

**Reply by jawalonoski** [reply]
Right now, no patient in Synthea misses wellness encounters, and everyone adheres to the vaccination schedule. Yes, you need to modify code to alter the wellness encounter schedule. Not sure how this relates to the JSON schema or JSON exporter though.

**Comment by garyisaac** [comment]
Thank you for the feedback. I can try using FHIR. I used the JSON export because it seemed easier than setting up a FHIR implementation. Drummond Group is launching a pediatric EHR testing program. I wanted to see if the well child visits followed recommended schedule for well child visits. We are testing some basic incomplete care opportunities reporting.

The system under test would need to filter patients based on missed, overdue, etc. well-child visits according to the schedule. I thought Synthea would be a good way to get test patients for reporting. I generated test patients using the 0-18 years old age group. I wanted to extract the well-child visits for a record and then manually verify it against the well-child visit schedule to see if I was on the right track.

Do you think this is a good approach? Or, would it make more sense to try to build a module to identify when well-child visits should occur according to the schedule? We are looking at creating 10-15 test patients. We are testing hundreds of patients at this point.

**Comment by jawalonoski** [comment]
> I can try using FHIR. I used the JSON export because it seemed easier than setting up a FHIR implementation.

The FHIR is written as JSON, so you could just handle it as JSON objects rather than in a fully-featured FHIR environment.

> Do you think this is a good approach? Or, would it make more sense to try to build a module to identify when well-child visits should occur according to the schedule? We are looking at creating 10-15 test patients. We are testing hundreds of patients at this point.

I'm not sure I have enough information. It seems reasonable that you could use the child wellness visits as test data for a program that identifies patients with missed or overdue visits. One thing you'd want to compare is whether or not the schedule used in Synthea is the same or not as the one used by the program under test, and identify the expect the behavior by comparing the two schedules. The schedule that Synthea uses is created from the `recommendedTimeBetweenWellnessVisits` method at the bottom of this post.

One thing you could do is generate different sets of patients after modifying the `recommendedTimeBetweenWellnessVisits` method to generate different visit patterns for your system to detect and flag.

https://github.com/synthetichealth/synthea/blob/5d90e086985aecf69e8a1140d9793ad8f104fe73/src/main/java/org/mitre/synthea/modules/EncounterModule.java#L173-L217
